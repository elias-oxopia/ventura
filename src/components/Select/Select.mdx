---
name: Select
menu: Components
route: /select
---

import { Playground, Props } from 'docz';
import { Select, Trash2} from '../../index.ts'

# Select

A Select can be used to let the user insert text using key strokes. A crucial form item for building interactive apps.

## Best practices

- Selects should be wrapped in a [FormField](/form-field) or should be preceded by another component serving as a label

## Examples

### Basic usage

<Playground>
  <Select isMulti options={[{label: <span><Trash2/>Bram</span>, value: 'test'},{label: "Test2", value: 'test2'}]} />
</Playground>

### Input with error

<Playground>
  <Select name="username" isInvalid />
</Playground>

## API

<Props of={Select} />
