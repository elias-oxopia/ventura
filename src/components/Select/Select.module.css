.select :global .ventura-select__control,
.select :global .ventura-select__control:hover {
  min-height: 36px;
  background-color: var(--color-neutral-1);
  border-radius: var(--border-radius-small);
  border: 1px solid var(--color-neutral-4);
}

.select :global .ventura-select__control--is-focused {
  outline: none;
  box-shadow: 0 0 0 3px var(--color-focus);
}

.select :global .ventura-select__placeholder {
  color: var(--color-neutral-6);
}

.select :global .ventura-select__value-container {
  padding-top: 0;
  padding-bottom: 0;
}

.select :global .ventura-select__menu {
  border: none;
  box-shadow: 0px 1px 5px rgba(0, 0, 0, 0.1);
}

.select :global .ventura-select__multi-value {
  height: 24px;
  background: var(--color-neutral-2);
  border-radius: 3px;
}

.select :global .ventura-select__multi-value__label {
  padding-top: 4px;
  padding-bottom: 3px;
  background: var(--color-neutral-2);
  border-radius: 3px;
  padding-left: 7px;
  color: var(--color-neutral-8);
}

.select :global .ventura-select__multi-value__remove {
  padding-right: 6px;
  padding-left: 4px;
  color: var(--color-neutral-5);
  cursor: pointer;
}

.select :global .ventura-select__multi-value__remove:hover {
  background-color: unset;
  color: unset;
}

.select :global .ventura-select__option {
  padding: 6px 12px;
}

.select :global .ventura-select__option--is-focused,
.select :global .ventura-select__option:active {
  background: var(--color-neutral-2);
}
