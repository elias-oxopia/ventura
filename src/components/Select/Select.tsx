import React, { ChangeEventHandler, FocusEventHandler } from 'react';
import Select, { OptionTypeBase } from 'react-select';
import classNames from 'classnames';

import cssReset from '../../css-reset.module.css';
import styles from './Input.module.css';

interface Props {
  options: OptionTypeBase[];
  value?: string;
  defaultValue?: string;
  onChange?: ChangeEventHandler<HTMLInputElement>;
  onBlur?: FocusEventHandler<HTMLInputElement>;
  className?: string;
}

const Input = React.forwardRef<Select, Props>(
  ({ options, value, defaultValue, onChange, onBlur, className }: Props, ref) => {
    const selectClassNames = classNames(cssReset.ventura, styles.input, className);

    const test = (t: any, actions: any) => {
      console.log(t, actions);
    };

    return (
      <Select
        ref={ref}
        option={options}
        inputValue={value}
        defaultInputValue={defaultValue}
        className={selectClassNames}
        onChange={test}
      />
    );
  },
);

export default Input;
