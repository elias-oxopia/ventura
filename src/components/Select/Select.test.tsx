import React, { ComponentProps } from 'react';
import { render } from '@testing-library/react';

import Select from './Select';

describe('Select', () => {
  const defaultSelectProps: ComponentProps<typeof Select> = {
    options: [
      { label: 'Option 1', value: '1' },
      { label: 'Option 2', value: '2' },
    ],
  };

  test('default snapshot', () => {
    const component = <Select {...defaultSelectProps} />;
    const { asFragment } = render(component);
    expect(asFragment()).toMatchSnapshot();
  });

  test('select with default value', () => {
    const component = (
      <Select {...defaultSelectProps} defaultValue={defaultSelectProps.options[0]} />
    );
    const { asFragment } = render(component);
    expect(asFragment()).toMatchSnapshot();
  });
});
