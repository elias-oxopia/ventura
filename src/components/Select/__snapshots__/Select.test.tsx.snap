// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select default snapshot 1`] = `
<DocumentFragment>
  <div
    class="ventura select css-2b097c-container"
  >
    <span
      aria-atomic="false"
      aria-live="polite"
      aria-relevant="additions text"
      class="css-1f43avz-a11yText-A11yText"
    />
    <div
      class="ventura-select__control css-yk16xz-control"
    >
      <div
        class="ventura-select__value-container css-g1d714-ValueContainer"
      >
        <div
          class="ventura-select__placeholder css-1wa3eu0-placeholder"
        />
        <div
          class="css-b8ldur-Input"
        >
          <div
            class="ventura-select__input"
            style="display: inline-block;"
          >
            <input
              aria-autocomplete="list"
              autocapitalize="none"
              autocomplete="off"
              autocorrect="off"
              id="react-select-2-input"
              spellcheck="false"
              style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
              tabindex="0"
              type="text"
              value=""
            />
            <div
              style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
            />
          </div>
        </div>
      </div>
      <div
        class="ventura-select__indicators css-1hb7zxy-IndicatorsContainer"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Select select with default value 1`] = `
<DocumentFragment>
  <div
    class="ventura select css-2b097c-container"
  >
    <span
      aria-atomic="false"
      aria-live="polite"
      aria-relevant="additions text"
      class="css-1f43avz-a11yText-A11yText"
    />
    <div
      class="ventura-select__control css-yk16xz-control"
    >
      <div
        class="ventura-select__value-container ventura-select__value-container--has-value css-g1d714-ValueContainer"
      >
        <div
          class="ventura-select__single-value css-1uccc91-singleValue"
        >
          Option 1
        </div>
        <div
          class="css-b8ldur-Input"
        >
          <div
            class="ventura-select__input"
            style="display: inline-block;"
          >
            <input
              aria-autocomplete="list"
              autocapitalize="none"
              autocomplete="off"
              autocorrect="off"
              id="react-select-3-input"
              spellcheck="false"
              style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
              tabindex="0"
              type="text"
              value=""
            />
            <div
              style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
            />
          </div>
        </div>
      </div>
      <div
        class="ventura-select__indicators css-1hb7zxy-IndicatorsContainer"
      />
    </div>
  </div>
</DocumentFragment>
`;
